app/controllers/books_controller.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
+  include Wor::Paginate
+
+  def index
+    @books = Book.all
+    render_paginated @books, each_serializer: BookSerializer
+  end
+
+  def show
+    @book = Book.find(params[:id])
```

Comment:
para esto lo óptimo sería "memoizarlo", puedes pegarle una leidita a esto para que entiendas mejor el concepto.
https://thoughtbot.com/blog/ruby-memoization-and-alternatives

Reply:
Además, ya que el show siempre va a necesitar un id para mostrar el record, deberías en 'requerir' el perametro, te recomiendo ver esta pregunta y su respectiva respuesta